sqlobject/src/main/java/org/jdbi/v3/sqlobject/internal/SqlAnnotations.java

```diff
@@ -35,36 +34,54 @@ private SqlAnnotations() {
     }
 
     /**
-     * Returns the <code>value()</code> of the <code>@SqlBatch</code>, <code>@SqlCall</code>, <code>@SqlQuery</code>, or
-     * <code>@SqlUpdate</code> annotation on the given method if declared and non-empty; empty otherwise.
+     * Returns the <code>value()</code> of the <code>@SqlBatch</code>, <code>@SqlCall</code>, <code>@SqlQuery</code>,
+     * <code>@SqlUpdate</code>, <code>@SqlScripts</code>, or <code>@SqlScript</code> annotation on the given method if declared and non-empty; empty otherwise.
      *
      * @param method the method
      * @return the annotation <code>value()</code>
      */
-    public static Optional<String> getAnnotationValue(Method method, Function<String, String> resolveSql) {
-        Predicate<String> nonEmpty = s -> !s.isEmpty();
+    public static Optional<String> getAnnotationValue(Method method) {
+        return getAnnotationValue(method, value -> value.isEmpty() ? null : value);
```

Comment:
The `nonEmpty` check moved here to keep filtering concerns out of the value lookup. Callers of the overloaded method sign up for having to do any filtering and replacements themselves.